### ISSUE TITLE
Generate TypeScript files from OpenAPI

### ISSUE THREAD

**Issue by pmcelhaney** [pmcelhaney/counterfact#45]
Once [TypeScript support](#44) is available we can start auto-generating a lot of the code from OpenAPI.

- Automatically stub out all of the files under routes.
- For each path, create a`*.types.ts` file that exports type signatures for request handler functions (GET, POST, etc.)
- The `*.types.ts` files are referenced by the path files 
- The `*.types.ts` files will reference another file or files that defines the schemas.
   - The schema files will probably be created with an existing [OpenAPI Generator](https://openapi-generator.tech/docs/generators)

```ts
// ./routes/user/[id]/hello.ts
import type { HTTP_GET  } from "./hello.types";

export const GET: HTTP_GET = ({query}) => {
  return {
    body: `Hello, ${query.name ?? "stranger"}`
  }
}
```

```ts
export type HTTP_GET_REQUEST = { query: { name?: string, store: any } }

export type HTTP_GET = (request: HTTP_GET_REQUEST) => { body: string } | { status: 201 } | { body: string[], status: 404 }
```

If a route file already exists, it will not overwrite. However `*.types.ts` files will be overwritten. We have no reason to edit those files manually.

**Comment by pmcelhaney** [comment]
This is "done" and I will merge the branch shortly, but there's a lot of technical debt. Code coverage went way down, lots of ESLint rules were ignored, and there's other silly stuff.

**Comment by pmcelhaney** [comment]
Released in version 0.5. It needs polish before 1.0 is released but it's working really well!
